Fix a bug in control

SvyatoslavLobov opened the pull request:
Fix a bug not updates the field "Property" of object "PropertyExpression" after deserialization "FilterScheme".

GeertvanHorrik commented on src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Models/FilterScheme.cs:
```diff
@@ -99,8 +99,9 @@ private void OnConditionItemsChanged()
-        protected override void OnDeserialized()
+        protected async override void OnDeserialized()
         {
+            await this.EnsureIntegrityAsync();
```
Not sure why you picked the async version inside a non-async method? Can you explain?

GeertvanHorrik replied:
I double checked, I see I made a mistake by marking the non-async obsolete. Please remove the ObsoleteEx() from the non-async method as well.

SvyatoslavLobov commented:
I picked the async version inside a non-async method, because non-async version was marked as ObsoleteEx.
Now I removed ObsoleteEx from non-async methods and rewrote the code using non-async version.
